code/service/src/main/scala/com/namely/chiefofstate/readside/ReadSideJdbcHandler.scala

```diff
@@ -56,18 +77,23 @@ private[readside] class ReadSideJdbcHandler(eventTag: String,
       // handle successful gRPC call where server indicated "successful = false"
       case Success(_) =>
         val errMsg: String =
-          s"read side returned failure, processor=${processorId}, id=${meta.entityId}, revisionNumber=${meta.revisionNumber}"
+          s"read side returned failure, attempt=${numAttempts}, processor=${processorId}, id=${meta.entityId}, revisionNumber=${meta.revisionNumber}"
         logger.warn(errMsg)
         throw new RuntimeException(errMsg)
 
       // handle failed gRPC call
       case Failure(exception) =>
         logger.error(
-          s"read side processing failure, processor=${processorId}, id=${meta.entityId}, revisionNumber=${meta.revisionNumber}, cause=${exception.getMessage()}"
+          s"read side processing failure, attempt=${numAttempts}, processor=${processorId}, id=${meta.entityId}, revisionNumber=${meta.revisionNumber}, cause=${exception.getMessage}"
         )
         // for debug purposes, log the stack trace as well
         logger.debug("remote handler failure", exception)
-        throw exception
+
+        val backoffSeconds: Long = Math.min(backOffSecondsMax, (backOffSecondsMin * Math.pow(1.1, numAttempts)).toLong)
+
+        Thread.sleep(Duration.ofSeconds(backoffSeconds).toMillis)
```

Comment:
Please can we avoid this?

Reply:
> Hey @AntonioYuen I don't think you need to implement this. You can just see how it is set because java grpc does have that feature.
> 
> * https://stackoverflow.com/questions/55428539/grpc-java-setting-enableretry-and-maxretryattempt-is-not-retrying
> * https://github.com/grpc/grpc-java/blob/master/core/src/main/java/io/grpc/internal/ExponentialBackoffPolicy.java
> * https://github.com/grpc/grpc/blob/master/doc/connection-backoff.md

Will look into it